CP charts endpoint

Aeilert opened the pull request:
Hi @tonyfujs ,

Here is a suggested endpoint for all the CP - charts (per country and poverty line). 

Maybe take and extra look at : 

* Selection of countries for Poverty MRV chart (see `cp_pov_mrv_select_countries()` and `test-ui_functions.R`) 
* Selection of year range for Poverty MRV chart 
* Whether we should have a single endpoint or mulitiple endpoints for the charts

tonyfujs commented on R/ui_functions.R:
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
+    lkup$svy_lkup[pcn_region_code == region]$country_code %>%
+    unique()
+
+  res_pov_mrv <- pip(country = countries, lkup = lkup)
```
Shouldn't this be:
`res_pov_mrv <- pip(country = countries, povline = povline, lkup = lkup)` Otherwise it will always use the `povline` default value and won't be sensitive to the poverty line chosen by the end-user.

Aeilert replied:
You are right. Good catch.

tonyfujs commented on R/ui_functions.R:
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
+    # IF selected country does not pertain to top 5 or bottom 5, display:
+    if (!(h %in% utils::tail(v, 5) | h %in% utils::head(v, 5))) {
+      v <- v[!v %in% h]
+      v2 <- v[v > h]
+      v3 <- v[v < h]
```
What do you think of replacing  
* `v2` with `v_above`, and
* `v3` with `v_below`

Aeilert replied:
Good idea!

tonyfujs commented on R/ui_functions.R:
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
+  if (nrow(dt) > 12) {
+    h <- dt[country_code == country]$headcount
+    v <- sort(dt$headcount)
+    # IF selected country does not pertain to top 5 or bottom 5, display:
+    if (!(h %in% utils::tail(v, 5) | h %in% utils::head(v, 5))) {
+      v <- v[!v %in% h]
+      v2 <- v[v > h]
+      v3 <- v[v < h]
+      vals <- c(h, utils::head(v, 3), utils::tail(v, 3),
+                utils::head(v2, 2), utils::tail(v3, 2))
+      # ELSE IF selected country pertains to top 5, display:
+    } else if (h %in% utils::tail(v, 5)) {
+      vals <- c(utils::head(v, 5), utils::tail(v, 6))
+      # ELSE display
+    } else {
+      vals <- c(utils::head(v, 6), utils::tail(v, 5))
+    }
```
I think I would probably extract this whole section into its own function. This is the heart of the selection algorithm, and it would be much easier to unit test edge cases.

Aeilert replied:
Also good idea.

tonyfujs reviewed (changes requested):
Hey @Aeilert 
Thanks for this. I added a few comments. Take a look and let me know what you think.

tonyfujs commented on R/ui_functions.R:
```diff
@@ -154,3 +167,140 @@ ui_cp_ki_headcount <- function(country, povline, lkup) {
+#' @export
+#'
+ui_cp_charts <- function(country = 'AGO', povline = NULL,
+                         pop_units = 1e6, year_range = 2016:2019,
+                         lkup) {
```
Shouldn't `year_range` be dynamic?

Aeilert replied:
Do you mean like a parameter in the API?

Aeilert commented:
> Hey @Aeilert
> Thanks for this. I added a few comments. Take a look and let me know what you think.

Hi @tonyfujs 

Looks good. I addressed your comments. And made a few other minor changes. 

Not sure what was up with the unit test for the number of indicators (intersection between pip() and /indicators), but I counted 20 (and not 19). With polarization now added to indicators_master.csv it is 21. 

All tests and checks pass, so I'm just going to go ahead and merge this.
